pkgs/development/libraries/vtk/generic.nix

```diff
@@ -51,7 +51,9 @@ in stdenv.mkDerivation rec {
     ] ++ optionals enablePython [
       pythonInterpreter
     ];
-  propagatedBuildInputs = optionals stdenv.isDarwin [ libobjc ];
+  propagatedBuildInputs = optionals stdenv.isDarwin [ libobjc ]
+    ++ optionals (stdenv.isLinux && lib.versionAtLeast majorVersion "9") [ libX11 libGL ];
```

Comment:
The expression for several versions is quite convoluted. I'd implement a policy that everything that goes towards support of the latest version and doesn't break older versions is just implemented unconditionally.

Reply:
Sure, I've propagated unconditionally, which is spurious for vtk_8 but hopefully fine.